doc/doxygen/scripts/program2doxygen

```diff
@@ -114,7 +128,27 @@ do {
             $state = $comment_mode;
         }
     }
-} while (<>);
+    elsif ($state == $code_fragment_mode)
+    {
+        # If this is the end of a @code - @endcode block, go back to
+        # comment_mode:
+        if (m!^\s*//\s*\@endcode!)
+        {
+            $state = $comment_mode;
+        }
+        # in code fragment mode: only skip leading whitespace and
+        # comment // signs
+        s!\s*//\s(.*)\n!$1!;
```

Comment:
I copied from what is already there. This is exactly how it is done a few lines above

Reply:
You mean in line 76 (old) / 90 (new): Yes, it's wrong there as well. We never noticed because it is unlikely that someone writes a comment of the kind
```
  // Now, it is necessary to recall that "//" indicates C++ comments.
```
whereas I can totally see us write a code block of the form
```
  // @code
  //       prm.add_parameter (y);    // declare an entry, and link it to a variable
  // @endcode
```

Reply:
Does it not work like sed regex, that the first match makes the regex expires, unless you specify to followup (`g`)? In this case, as long as a comment is used at the beginning of the line, the rest is marked as a match (all of it, including the other potential match)

Reply:
That's probably correct. The only place where it wouldn't work is if it encounters a comment of the form
```
  //some text with a missing space at the beginning talking about // tokens
```